pkg/trtl/replica.go

```diff
@@ -85,13 +419,264 @@ func (r *ReplicaService) SelectPeer() (peer *peers.Peer) {
 	return nil
 }
 
-// During gossip, the initiating replica sends a randomly selected remote peer the
-// version vectors of all objects it currently stores. The remote peer should
-// respond with updates that correspond to more recent versions of the objects. The
-// remote peer can than also make a reciprocal request for updates by sending the
-// set of versions requested that were more recent on the initiating replica, and
-// use a partial flag to indicate that it is requesting specific versions. This
-// mechanism implements bilateral anti-entropy: a push and pull gossip.
-func (r *ReplicaService) Gossip(ctx context.Context, in *replica.VersionVectors) (out *replica.Updates, err error) {
-	return nil, status.Error(codes.Unimplemented, "this replica does not yet implement gossip")
+// Gossip implements biltateral anti-entropy: during a Gossip session the initiating
+// replica pushes updates to the remote peer and pulls requested changes. Using
+// bidirectional streaming, the initiating peer sends data-less sync messages with
+// the versions of objects it stores locally. The remote replica then responds with
+// data if its local version is later or sends a sync message back requesting the
+// data from the initating replica if its local version is earlier (no exchange)
+// occurs if both replicas have the same version. At the end of a gossip session,
+// both replicas should have synchronized and have identical underlying data stores.
+func (r *ReplicaService) Gossip(stream replica.Replication_GossipServer) (err error) {
```

Comment:
Is there some way we can better clarify that the `AntiEntropySync` is the "client" method and the `Gossip` is the "server" method in this interaction? Perhaps they should be in separate files? The way I'm thinking about this is that there is one set of things that the initiator does and another set of things that the remote does and separating those two roles makes sense to me.

Reply:
I'll think about it in sc-2694 and add your comment there.